src/doc/rustdoc/src/advanced-features.md

```diff
@@ -110,3 +110,23 @@ https://doc.rust-lang.org/stable/std/?search=%s&go_to_first=true
 
 This URL adds the `go_to_first=true` query parameter which can be appended to any `rustdoc` search URL
 to automatically go to the first result.
+
+## `#[repr(transparent)]`: Documenting the transparent representation
+
+You can read more about `#[repr(transparent)]` itself in the [Rust Reference][repr-trans-ref] and
+in the [Rustonomicon][repr-trans-nomicon].
+
+Since this representation is only considered part of the public ABI if the single field with non-trivial
+size or alignment is public and if the documentation does not say otherwise, Rustdoc displays the
+attribute if and only if the non-1-ZST field is public or at least one field is public in case all
+fields are 1-ZST fields. The term *1-ZST* refers to types that are one-aligned and zero-sized.
+
+If you would like to declare the representation as private even if the non-1-ZST field is public,
+document it in prose. It would seem that you can manually hide the attribute with
+`#[cfg_attr(not(doc), repr(transparent))]`. However, due to [current limitations][cross-crate-cfg-doc],
+this does not work for types re-exported from another crate. Therefore, you should always additionally
+explicitly write down the layout guarantees in such a case.
```

Comment:
I'm pretty sure the phrasing in this section can be improved upon, feels a bit awkward in some places.